droidlet/tools/hitl/nsp_retrain_infra.py

```diff
@@ -161,28 +161,39 @@ def create_mask(self, opts, batch_id, data_filepath):
         # Create train, valid, and test masks based on user input
         total_rows = sum(1 for line in open(data_filepath))
         logging.info(f"Model training data masks are being generated:")
-        if  not opts.resample:
-            logging.info(f"Model will be trained with old+new training data, valid/test are static old data (not resampled)")
-
-            # Lengthen valid and test masks to be the length of all data
-            old_mask_filepath = os.path.join(opts.sweep_config_folder, 'split_masks.pth')
-            old_masks = torch.load(old_mask_filepath)['annotated']
-            valid_mask = [x for x in old_masks['valid']]
-            valid_mask.extend([False] * (total_rows - len(valid_mask)))
-            test_mask = [x for x in old_masks['test']]
-            test_mask.extend([False] * (total_rows - len(test_mask)))
-            train_mask = [False if valid_mask[i] or test_mask[i] else True for i in range(total_rows)]  # Rest of the data are train
-        else:
-            logging.info(f"Model train/valid/test data will be resampled from combined old+new dataset in 80/10/10 split")
-
-            # Generate random splits and populate masks
-            mask_designation = ['test'] * int(total_rows * 0.1)
-            mask_designation.extend(['valid'] * int(total_rows * 0.1))
-            mask_designation.extend(['train'] * (total_rows - len(mask_designation)))
-            random.shuffle(mask_designation)
-            train_mask = [True if x == 'train' else False for x in mask_designation]
-            valid_mask = [True if x == 'valid' else False for x in mask_designation]
-            test_mask = [True if x == 'test' else False for x in mask_designation]
+
+        # Old masks are static, just extend to be the length of the new dataset
+        old_mask_filepath = os.path.join(opts.sweep_config_folder, 'split_masks.pth')
+        old_masks = torch.load(old_mask_filepath)['annotated']
+        old_train_mask = [x for x in old_masks['train']]
+        old_train_mask.extend([False] * (total_rows - len(old_train_mask)))
+        old_valid_mask = [x for x in old_masks['valid']]
+        old_valid_mask.extend([False] * (total_rows - len(old_valid_mask)))
+        old_test_mask = [x for x in old_masks['test']]
+        old_test_mask.extend([False] * (total_rows - len(old_test_mask)))
+
+        # Set random seed for reproducability and generate masks for new data
+        random.seed(batch_id)
+        new_mask_designation = ['test'] * int(new_data_rows * 0.1)
+        new_mask_designation.extend(['valid'] * int(new_data_rows * 0.1))
+        new_mask_designation.extend(['train'] * (new_data_rows - len(new_mask_designation)))
+        random.shuffle(new_mask_designation)
```

Comment:
can we just write a function for this instead of redundant steps here ?
or something less redundant like:
```
all_masks = {}
for mask_type in ['test', 'train', 'valid']:
    all_masks[mask_type] = [False] * len(new_masks)
    for j in [i for i in range(len(new_masks)) if new_masks[i] == mask_type]:
        all_masks[mask_type][j] = True
```

Reply:
For sure, let me know if you like the way I rebuilt things in the latest commit.